tests/bofire/strategies/test_mobo.py

```diff
@@ -0,0 +1,214 @@
+)
+from botorch.acquisition.multi_objective.logei import (  # qExpectedHypervolumeImprovement,
+    qLogExpectedHypervolumeImprovement,
+    qLogNoisyExpectedHypervolumeImprovement,
+)
+
+# qNoisyExpectedHypervolumeImprovement,
+from botorch.acquisition.multi_objective.objective import GenericMCMultiOutputObjective
+
+import bofire.data_models.acquisition_functions.api as acquisitions
+import bofire.data_models.strategies.api as data_models
+import bofire.strategies.api as strategies
+from bofire.benchmarks.multi import C2DTLZ2, DTLZ2
+from bofire.data_models.features.api import ContinuousOutput
+from bofire.data_models.strategies.api import (
+    PolytopeSampler as PolytopeSamplerDataModel,
+)
+from bofire.strategies.api import PolytopeSampler
+from tests.bofire.utils.test_multiobjective import (
+    dfs,
+    invalid_domains,
+    valid_constrained_domains,
+    valid_domains,
+)
+
+
+@pytest.mark.parametrize(
+    "domain, ref_point",
+    [
+        (invalid_domains[0], None),
+        (invalid_domains[1], None),
+        (valid_domains[0], [0]),
+        (valid_domains[0], {}),
+        (valid_domains[0], {"of1": 0.0, "of2": 0, "of3": 0}),
+        (valid_domains[0], {"of1": 0.0}),
+        (valid_domains[0], {"of1": 0.0, "of3": 0.0}),
+    ],
+)
+def test_invalid_mobo(domain, ref_point):
+    with pytest.raises(ValueError):
+        data_models.MoboStrategy(domain=domain, ref_point=ref_point)
+
+
+@pytest.mark.parametrize("domain", valid_constrained_domains)
+def test_qnehvi_valid_constrained_objectives(domain):
+    data_models.MoboStrategy(domain=domain)
+
+
+@pytest.mark.parametrize(
+    "domain, ref_point, experiments, expected",
+    [
+        (valid_domains[0], {"of1": 0.5, "of2": 10.0}, dfs[0], [0.5, -10.0]),
+        (valid_domains[1], {"of1": 0.5, "of3": 0.5}, dfs[1], [0.5, 0.5]),
+        (valid_domains[0], None, dfs[0], [1.0, -5.0]),
+        (valid_domains[1], None, dfs[1], [1.0, 2.0]),
+    ],
+)
+def test_mobo_get_adjusted_refpoint(domain, ref_point, experiments, expected):
+    data_model = data_models.MoboStrategy(domain=domain, ref_point=ref_point)
+    strategy = strategies.map(data_model)
+    # hack for the test to prevent training of the model when using tell
+    strategy.set_experiments(experiments)
+    adjusted_ref_point = strategy.get_adjusted_refpoint()
+    assert isinstance(adjusted_ref_point, list)
+    assert np.allclose(expected, np.asarray(adjusted_ref_point))
+
+
+@pytest.mark.parametrize(
+    "strategy, use_ref_point, acqf, num_test_candidates",
+    [
+        (data_models.MoboStrategy, use_ref_point, acqf, num_test_candidates)
+        for use_ref_point in [True, False]
+        for acqf in [
+            acquisitions.qEHVI,
+            acquisitions.qLogEHVI,
+            acquisitions.qNEHVI,
+            acquisitions.qLogNEHVI,
+        ]
+        for num_test_candidates in range(1, 3)
+    ],
+)
+def test_mobo(strategy, use_ref_point, acqf, num_test_candidates):
```

Comment:
is num_test_candidates being used in this function?

Reply:
Good point, this is an copy paste thing, I will remove it.

Reply:
Fixed it.